core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -548,42 +556,47 @@ public RequestLogBuilder logBuilder() {
 
     @Override
     public String toString() {
-        String strVal = this.strVal;
         if (strVal != null) {
             return strVal;
         }
+        return toStringSlow();
 
-        final StringBuilder buf = new StringBuilder(107);
-        buf.append("[C]");
+    }
 
-        // Prepend the current channel information if available.
+    private String toStringSlow() {
+        // Prepare all properties required for building a string representation,
```

Comment:
This is less clear again. I think we have a convention of not calling any methods during thread local building because it's not clear whether that method may reinitialize the buffer or not. toString is of course a common culprit but not necessarily the only one.

Actually the javadoc rule of thumb doesn't seem strong enough, it's not really an issue of recursion and just about whether other methods may initialize a buffer so the general rule of thumb is to call no other methods unless they accept a `StringBuilder` as a parameter.
https://github.com/line/armeria/blob/9061e241961c806951dd855be65fd871131a1835/core/src/main/java/com/linecorp/armeria/internal/common/util/TemporaryThreadLocals.java#L47

Reply:
Updated again, this time `TemporaryThreadLocals` as well.